src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Party;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+using static System.Reflection.BindingFlags;
+using static CommunityPatch.PatchApplicabilityHelper;
+
+namespace CommunityPatch.Patches.Perks.Social.Leadership {
+
+  public sealed class UltimateLeaderPatch : PatchBase<UltimateLeaderPatch> {
```

Comment:
There's also `PerkPatchBase<>` you could use now.
Might need to be updated to handle the scenario of duplicate perk IDs, but it _might_ work ok.